### ISSUE TITLE
Index 0 out of bounds for length 0 -- Check script 'modules/local/jupyternotebook/./parametrize.nf' at line: 27

### ISSUE THREAD

**Issue by vjmarteau** [grst/nextflow-notebook-pipeline#1]
Running: `NXF_VER=23.04.0 nextflow run ./main.nf` seems to be the last nextflow version that runs without the above error.

**Comment by grst** [comment]
Yeah, this was also brought up in the `nmp-liver` project: https://github.com/icbi-lab/nmp-liver/commit/0036ef0a7a1bb10fa22734610ff96f44f226a0df

The parametrize script needs some updating, also in nf-core modules. I didn't have time to look into it yet.

**Comment by vjmarteau** [comment]
https://github.com/grst/nextflow-notebook-pipeline/blob/e9c2b4bcd915f61090addff5744f07a9accf9586/modules/nf-core/modules/jupyternotebook/parametrize.nf#L27-L29

The Index 0 out of bounds error seems to be caused by the new Groovy v3 parser and is fixed by removing the curly brackets. Also, newer versions of Nextflow use snakeyaml v2 which needs DumperOptions to be passed explicitly to the representer. The changes below work for me.

```groovy
def representer = new Representer(options)
representer.multiRepresenters.put(GString, representer.representers.get(String))
```
